Qt.py

```diff
@@ -783,14 +783,30 @@ def _translate(context, sourceText, *args):
     # The first argument is disambiguation[str]
     # The last argument is n[int]
     # The middle argument can be encoding[QtCore.QCoreApplication.Encoding]
-    if len(args) == 3:
-        disambiguation, encoding, n = args
-    elif len(args) == 2:
-        disambiguation, n = args
-        encoding = None
-    else:
+    disambiguation = encoding = n = None
+    args = list(args)
+
+    # First arg needs to be (str, NoneType)
+    # Last arg needs to be (int, NoneType)
+    # If last arg is "n", we require "disambiguation" to be passed
+    #    In Qt4, if "n" is passed, encoding must be passed.
+    #    In Qt5
+    if len(args) > 3:
         raise TypeError(
-            "Expected 4 or 5 arguments, got {0}.".format(len(args)+2))
+                "Expected 4 or 5 arguments, got {0}.".format(len(args)+2)
```

Comment:
Surprised the Hound didn't point this out, but there should be spaces around the `+` sign.